### ISSUE TITLE
constant() should support dynamic args

### ISSUE THREAD

**Issue by ross-nordstrom** [caolan/async#1016]
In `v1.5.2` constant seems to be hard-coded to be invoked with 1 arg: a callback. **This is limiting**

```js
    async.constant = _restParam(function(values) {
        var args = [null].concat(values);
        return function (callback) {
            return callback.apply(this, args);
        };
    });
```

I'd like to propose the inner, curried function take dynamic args as well:

```js
    async.constant = _restParam(function(values) {
        var args = [null].concat(values);
        return _restParam(function (callbackArgs) { // <= We only care about the callback,
            var callback = callbackArgs.pop();      //    and assume it's the last arg
            return callback.apply(this, args);      //    when the curried function is invoked
        });
    });
```

## Why do I care?

Composability.  I don't want to care about the callback-signature of other functions in Async-chains.

```js
async.waterfall([
  doSomething/*(cb)*/, //=> Invokes cb with (err, arg1, arg2)
  async.constant('Pretend doSomething did somethingElse')/*(arg1, arg2, cb)*/
], doneCb);
```

**Comment by aearly** [comment]
I like this -- a simple enhancement.

**Comment by aearly** [comment]
Closed via #1052
